department-of-veterans-affairs/caseflow
# Craig/appeals 13253

Resolves [APPEALS-13253](https://vajira.max.gov/browse/APPEALS-13253)

### Description
Functionality:
- Added ActiveRecord association between cavc_remand and cavc_dashboard_dispositions
- Added attribute cavc_dashboard_dispositions to cavc_remand_serializer
- Modified cavc_dashboard_controller#index format.json to search for a CavcRemand by source and remand appeal UUID, then get all other CavcRemands that share a source or remand appeal with that CavcRemand and return that to the frontend
- Modified cavcDashboard.jsx and its reducer and actions files for the backend data changes
- Changed cavcDashboard.jsx redirect to check for the presence of CavcRemands from the new backend methods instead of checking the appealDetails
- Added a boolean to TabWindow.jsx that will allow it to always render visual tabs (it was configured to not render the tabs if only one tab was passed in)
- Added component CavcDashboardTab which will contain the other sections of the Cavc Dashboard
- Added TabWindow to cavcDashboard.jsx which accepts the CavcDashboardTab component for rendering

Tests and Seed Data:
- Added test to TabWindow.test.js for rendering visual tabs with new boolean, and updated snapshots for new boolean
- Added seed data which creates a single appeal with four associated CavcRemands
- Added/fixed tests for new seed data
- Modified cavc_dashboard_controller_spec for changes to controller
- Modified unrecognized_appellant_spec to fix flaky test

Other:
- Modified AppFrame component's propTypes to allow wideApp to be either String or Boolean; addresses errors in the console for unexpected propType

### Acceptance Criteria

### Testing Plan
1. Go to demo environment
2. Go to search page
3. Search for a veteran with multiple CAVC remands (410000011)
4. Go to case details page for the **original** appeal stream
5. In the URL for the case details page, add "/cavc_dashboard" and go to that page
6. Verify that four CAVC dashboard tabs are displayed that match the wireframes
7. Verify that each tab can be clicked, and that the CAVC docket number changes in the content area and matches the CAVC docket number on the tab
8. Go back to search
9. Search for a veteran with a single CAVC remand (200000250)
10. Go to case details page for the **CAVC** appeal stream
11. In the URL for the case details page, add "/cavc_dashboard" and go to that page
12. Verify that a single CAVC dashboard tab is displayed that matches the wireframes

### Code Documentation Updates
